### PR TITLE
Websites from Legacy to Haru #1

### PR DESCRIPTION
**Pull request by MikeZeDev**
* AGCScanlation
* ASMHentai
* AsuraScansTR
* BlogTruyen

### CODE REVIEW

File: src/engine/websites/AGCScanlation.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import * as Common from './decorators/Common';
+
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    const title = anchor.querySelector('.nomeserie span').textContent.trim();
+    const id = anchor.pathname + anchor.search;
+    return { id, title };
+}
+
+function ChapterInfoExtractor(anchor: HTMLAnchorElement) {
+    const title = anchor.text;
```
Comment (ronny1982) [inline src/engine/websites/AGCScanlation.ts:13]: Is `trim()` not necessary?

**Comment by ronny1982** [comment]
Was the change of permissions for the icons from 0755 to 0644 by intention?

File: src/engine/websites/AsuraScansTR.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@MangaStream.MangaCSS(/^https?:\/\/asurascanstr\.com\/manga\/[^/]+\/$/)
+@MangaStream.MangasSinglePageCSS()
+@MangaStream.ChaptersSinglePageCSS()
+@MangaStream.PagesSinglePageCSS([/ENDING-PAGE/i], 'div#readerarea p img')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('asurascans-tr', `Asura Scans (TR)`, 'https://asurascanstr.com', Tags.Language.Turkish, Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Media.Manga);
```
Comment (ronny1982) [inline src/engine/websites/AsuraScansTR.ts:16]: May add Scanlator tag

File: src/engine/websites/AsmHentai.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+@Common.MangaCSS(/^https?:\/\/asmhentai\.com\/g\/[^/]+\/$/, 'div.book_page div.info h1')
+@Common.MangasNotSupported()
+@MangaStream.PagesSinglePageJS([], script)
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('asmhentai', `AsmHentai`, 'https://asmhentai.com', Tags.Language.Multilingual, Tags.Rating.Erotica);
```
Comment (ronny1982) [inline src/engine/websites/AsmHentai.ts:32]: May add Aggregator tag

File: src/engine/websites/BlogTruyen.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@Common.MangaCSS(/^https?:\/\/blogtruyen\.vn\/[^/]+\//, 'section.manga-detail h1.entry-title')
+@Common.MangasMultiPageCSS('/ajax/Search/AjaxLoadListManga?key=tatca&orderBy=1&p={page}', 'div.list p span.tiptip a', 1, 1, 200)
+@Common.ChaptersSinglePageCSS('div#list-chapters span.title a')
+@Common.PagesSinglePageCSS('article#content img')
+@Common.ImageDirect()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('blogtruyen', `BlogTruyen`, 'https://blogtruyen.vn', Tags.Language.Vietnamese, Tags.Media.Manhua, Tags.Media.Manhwa);
```
Comment (ronny1982) [inline src/engine/websites/BlogTruyen.ts:15]: May add Aggregator tag

File: src/engine/websites/BlogTruyen_e2e.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        url: 'https://blogtruyen.vn/15325/arcadia-of-the-moonlight',
+        id: '/15325/arcadia-of-the-moonlight',
+        title: 'Arcadia Of The Moonlight'
+    },
+    child: {
+        id: '/c247578/arcadia-of-the-moonlight-chap-1',
+        title: 'chap 1'
+    },
+    entry: {
+        index: 0,
```
Comment (ronny1982) [inline src/engine/websites/BlogTruyen_e2e.ts:18]: [This image](https://blogtruyen.vn/donate.png) looks very generic, probably better to use an image that only belongs to this chapter under test e.g., image 5.
The first images seems to be bloat, maybe these can be removed.

**Comment by MikeZeDev** [comment]
> Was the change of permissions for the icons from 0755 to 0644 by intention?

I didnt do such thing. I work on windows

File: src/engine/websites/BlogTruyen_e2e.ts (hunk 5 above)
Comment (MikeZeDev) [reply src/engine/websites/BlogTruyen_e2e.ts:18]: done

**Review by ronny1982 — APPROVED** [review]

**Pull request merged**